### PR TITLE
🐙 source-dynamodb: fix incorrect type check for incremental read

### PR DESCRIPTION
**Pull request by laskoviymishka**
## What

Dynamodb connector only allow us to do incremental on string / nubmer columns, which is stated in docs. But it can’t produce such column type, see [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbSchemaSerializer.java#L34) for numbers and here for [strings](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbSchemaSerializer.java#L22), so in a fact column type is not a string, but one of null / string (i.e. [“null”, “string”]). But code that verifies this, actually expects just “string”, see [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbSource.java#L154), this leads us to error that we observed here: #41681.

## How

After some consideration I decide to make changes as minimal as possible, I altered the way how dynamodb type infered from airbyte type, added to just `string` or `integer` also optional `["null","string"]`.

## Review guide
Main changes [here](https://github.com/airbytehq/airbyte/blob/b08eae6a31f864627d32cc67dbb83f18a65c5117/airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbSource.java#L151)

## User Impact

No changes in API / Output, incremental read start to work.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbSource.java [hunk 1]
```diff
@@ -146,14 +146,15 @@ private AutoCloseableIterator<AirbyteMessage> scanIncremental(final DynamodbOper
 
     // cursor type will be retrieved from the json schema to save time on db schema crawling reading
     // large amount of items
-    final String cursorType = properties.get(cursorField).get("type").asText();
+    final String cursorType = properties.get(cursorField).get("type").toString();
+    LOGGER.info("cursor type: {}", cursorType);
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbSource.java:150]: Does the rest of the source use this logging pattern, or a different one?

**Review by natikgadzhi — COMMENTED** [review]
Thank you for this! I know DB sources team is more backlogged on OSS contributions, but hope @marcosmarxm and I can convince them to take a quick look ;)

File: airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbSource.java (hunk 1 above)
Comment (laskoviymishka) [reply airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbSource.java:150]: as far as i see - yes

File: airbyte-integrations/connectors/source-dynamodb/metadata.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: 50401137-8871-4c5a-abb7-1f5fda35545a
-  dockerImageTag: 0.3.2
+  dockerImageTag: 0.3.3
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-dynamodb/metadata.yaml:8]: Can you bump the doc changelog entry too? Then we're going to be able to run CI integration tests to validate your change.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
One comment.

File: airbyte-integrations/connectors/source-dynamodb/metadata.yaml (hunk 2 above)
Comment (laskoviymishka) [reply airbyte-integrations/connectors/source-dynamodb/metadata.yaml:8]: done

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**